homeassistant/components/nest/media_source.py

```diff
@@ -56,6 +60,177 @@
+    of individual objects stored here (e.g. purging when going over storage
+    limits). This store manages the addition/deletion once instructed.
+    """
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        subscriber: GoogleNestSubscriber,
+        store: Store,
+        media_path: str,
+    ) -> None:
+        """Initialize NestEventMediaStore."""
+        self._hass = hass
+        self._subscriber = subscriber
+        self._store = store
+        self._media_path = media_path
+        self._data: dict | None = None
+        self._data_loaded = False
+        self._devices: Mapping[str, str] | None = {}
+
+    async def async_load(self) -> dict | None:
+        """Load data."""
+        if not self._data_loaded:
+            self._devices = await self._get_devices()
+            data = await self._store.async_load()
+            if data is None:
+                _LOGGER.debug("Loaded empty event store")
+                self._data = {}
+            elif isinstance(data, dict):
+                _LOGGER.debug("Loaded event store with %d records", len(data))
+                self._data = data
+            else:
+                _LOGGER.error(
+                    "Unexpected data in storage version=%s, key=%s",
+                    STORAGE_VERSION,
+                    STORAGE_KEY,
+                )
+            self._data_loaded = True
+        return self._data
+
+    async def async_save(self, data: dict | None) -> None:
+        """Save data."""
+        self._data = data
+
+        def provide_data() -> dict:
+            return {} if data is None else data
+
+        self._store.async_delay_save(provide_data, STORAGE_SAVE_DELAY_SECONDS)
+
+    def get_media_key(self, device_id: str, event: ImageEventBase) -> str:
+        """Return the filename to use for a new event."""
+        # Convert a nest device id to a home assistant device id
+        device_id_str = (
+            self._devices.get(device_id, "unknown_device")
+            if self._devices
+            else "unknown_device"
+        )
+        event_id_str = event.event_session_id
+        try:
+            raise_if_invalid_filename(event_id_str)
+        except ValueError:
+            event_id_str = ""
+        time_str = str(int(event.timestamp.timestamp()))
+        event_type_str = EVENT_NAME_MAP.get(event.event_type, "event")
+        suffix = "jpg" if event.event_image_type == EventImageType.IMAGE else "mp4"
+        return f"{device_id_str}/{time_str}-{event_id_str}-{event_type_str}.{suffix}"
+
+    def get_media_filename(self, media_key: str) -> str:
+        """Return the filename in storage for a media key."""
+        return f"{self._media_path}/{media_key}"
+
+    async def async_load_media(self, media_key: str) -> bytes | None:
+        """Load media content."""
+        filename = self.get_media_filename(media_key)
+
+        def load_media(filename: str) -> bytes | None:
+            if not os.path.exists(filename):
```

Comment:
filename is not prefixed with the config dir ?

Reply:
Ah nm, see now it's using `self._media_path`.

Reply:
I think this is handled by `get_media_filename` which has `_media_path` as a prefix which is set up by `async_get_media_event_store`.    perhaps a different variable name could make this more clear? e.g. `media_full_path =`?